native/cocos/base/StringPool.h

```diff
@@ -35,7 +35,7 @@
 
 namespace cc {
 
-namespace {
+namespace impl {
```

Comment:
Why need a namespace name? This name is easy to conflict.

Reply:
gcc reports error here
```
/home/pt/Github/engine/native/cocos/base/StringPool.h: In instantiation of ‘class cc::StringPool<false>’:

/home/pt/Github/engine/native/cocos/renderer/frame-graph/PassInsertPointManager.h:53:16:   required from here

/home/pt/Github/engine/native/cocos/base/StringPool.h:55:7: error: ‘cc::StringPool<false>’ has a field ‘cc::StringPool<false>::_stringToHandles’ whose type uses the anonymous namespace [-Werror=subobject-linkage]
```

Reply:
The warning `-subobject-linkage` has been ignored.

Reply:
The definitions of StringEqual and StringHash have been moved to the inside of the StringPool class.